src/sharding.js

```diff
@@ -109,7 +159,7 @@ class ShardingDatastore extends Adapter {
 
     if (q.prefix != null) {
       tq.filters.push((e) => {
-        return this._invertKey(e.key).toString().startsWith(q.prefix)
+        return this._invertKey(e.key).toString().startsWith(q.prefix ? q.prefix : '')
```

Comment:
We do know `prefix` is there because of the if clause. However TS does not ,because `tq.filter` or `this._invertKey` or other things that happen here could be mutating `q.prefix`. With the suggestion above following could be simplified as proposed

```suggestion
        return this._invertKey(e.key).toString().startsWith(prefix)
```